test/e2e/framework/helper/poll.go

```
@@ -125,34 +111,33 @@ func (h *Helper) WaitForDeploymentToDelete(namespace, name string, timeout time.
 	})
 
 	if err != nil {
-		kErr := h.Kubectl(namespace).DescribeResource("deployment", name)
-		if kErr != nil {
+		if kErr := h.Kubectl(namespace).DescribeResource("deployment", name); kErr != nil {
 			err = fmt.Errorf("%s\n%s", err, kErr)
 		}
-
-		return err
 	}
+	return err
 
-	return nil
 }
 
-func (h *Helper) WaitForUrlToBeReady(url *url.URL, timeout time.Duration) error {
-	log.Infof("Waiting for URL %s to be ready", url)
+func (h *Helper) WaitForUrlToBeReady(targetURL *url.URL, timeout time.Duration) error {
+	log.Infof("Waiting for URL %s to be ready", targetURL)
 
-	err := wait.PollImmediate(time.Second*2, timeout, func() (bool, error) {
-		host := url.Host
-		port := url.Port()
-		tocheck := host
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
 
-		if port != "" {
-			tocheck = tocheck + ":" + port
+	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+		host := targetURL.Host
+		if port := targetURL.Port(); port != "" {
+			host = host + ":" + port
 		}
 
-		con, err := net.DialTimeout("tcp", tocheck, timeout)
+		con, err := net.DialTimeout("tcp", host, timeout)
 		if err != nil {
 			return false, nil
 		} else {
-			con.Close()
+			defer func() {
+					_ = con.Close()
```

Don't ignore the error.